### PR TITLE
HWKAPM-629 Fixed missing propagation of the trace id between spawned …

### PR DESCRIPTION
**Pull request by objectiser**
…fragments, which was causing misleading results in the distributed trace diagram

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay Could you review this please?

File: client/collector/src/test/java/org/hawkular/apm/client/collector/internal/DefaultTraceCollectorTest.java [hunk 1]
```diff
@@ -842,6 +843,29 @@ public void testSpawnFragmentUsingAddChild() {
     }
 
     @Test
+    public void testSpawnFragmentSameTraceId() {
+        DefaultTraceCollector collector = new DefaultTraceCollector();
+
+        FragmentBuilder parent = new FragmentBuilder();
+        FragmentBuilder spawned = new FragmentBuilder();
+
+        Trace parentBTxn = parent.getTrace();
```
Comment (pavolloffay) [inline client/collector/src/test/java/org/hawkular/apm/client/collector/internal/DefaultTraceCollectorTest.java]: Why `BTxn` suffix? Isn't `parentTrace` more intuitive?

File: client/collector/src/test/java/org/hawkular/apm/client/collector/internal/DefaultTraceCollectorTest.java [hunk 2]
```diff
@@ -842,6 +843,29 @@ public void testSpawnFragmentUsingAddChild() {
...
+        Trace spawnedBTxn = spawned.getTrace();
```
Comment (pavolloffay) [inline client/collector/src/test/java/org/hawkular/apm/client/collector/internal/DefaultTraceCollectorTest.java]: The same as above

**Comment by objectiser** [comment]
@pavolloffay Thanks! Updated applied.

**Pull request merged**